src/Examples/UnitModels/Operations/skeleton_unit_testing.ipynb

```diff
@@ -0,0 +1,420 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# IDAES Skeleton Unit Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook demonstrates usage of the IDAES Skeleton Unit Model, which provides a generic \"bares bones\" unit for user-defined models and custom variable and constraint sets. To allow maximum versatility, this unit may be defined as a surrogate model or a custom equation-oriented model. Users must add ports and variables that match connected models, and this is facilitated through a provided method to add port-variable sets.\n",
+    "\n",
+    "For users who wish to train surrogates with IDAES tools and insert obtained models into a flowsheet, see the more versatile `SurrogateBlock()` for streamlined integration."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will begin with relevant imports. We will need basic Pyomo and IDAES components:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pytest\n",
+    "from pyomo.environ import (check_optimal_termination,\n",
+    "                           ConcreteModel,\n",
+    "                           SolverFactory,\n",
+    "                           Constraint,\n",
+    "                           Var,\n",
+    "                           Set,\n",
+    "                           value,\n",
+    "                           units as pyunits)\n",
+    "from idaes.core import FlowsheetBlock\n",
+    "from idaes.models.unit_models import SkeletonUnitModel\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "from pyomo.util.check_units import assert_units_consistent"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For demonstrative purposes, we will build a simple model manually defining state variables relations for two generic components. In this model, the variables specify an FcTP system where molar flow of each component, temperature and pressure are selected as state variables. Users can use this structure to write custom relations between inlet and outlet streams; for example, defining outlet pressure using an explicit correlation or outlet flow of a specific component from a known accumulation or loss in the system."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = ConcreteModel()\n",
```

Comment:
This needs to be broken up, with descriptions of each part explaining what is being done and why and referring back to the introduction to the model.